### DISCUSSION TITLE
Correct use of Prisma.$disconnect

### DISCUSSION THREAD

**Discussion by AndryHTC** [medz/dart-orm#448] [Q&A]
It's not clear when disconnect should be called. 
Is it something to do after each query/groups of queries?

Can avoiding this cause idle connections?

I'm using Serverpod

**Comment by medz** [comment]
No, you don't need to actively disconnect. Because there is a connection pool mechanism behind the binary engine, there will only be one connection on standby when it is inactive. You should disconnect it when your server terminates, or you need to temporarily open a new connection and close it when you are done with it.

Therefore, in general. You should not disconnect the database connection, but you should disconnect it when your server stops (although it will automatically disconnect, it is not stable).
